src/Lean/Meta/Eqns.lean

```diff
@@ -136,6 +136,18 @@ private partial def alreadyGenerated? (declName : Name) : MetaM (Option (Array N
   else
     return none
 
+/--
+Equation theorems to use instead of the generated ones.
+-/
+builtin_initialize userEqnsExt : MapDeclarationExtension (Array Name) ← mkMapDeclarationExtension
+
+/--
+Adds the given equation theorems to the user equations extension.
+-/
+def setUserEqns (declName : Name) (eqns : Array Name) : CoreM Unit := do
+  registerEqnThms declName eqns
```

Comment:
Oops, there's a bug here. The point of calling this is to clear the cached equation theorems, but `registerEqnThms` does not expect being called on a declaration that has already had equation theorems registered. The issue is that `mapInv` is still going to say that the previous equation theorems are equation theorems, making `isEqnThm?` provide incorrect results.